backend.native/tests/interop/objc/tests/kt41811.kt

```diff
@@ -81,3 +81,29 @@ private fun createGarbageDeallocLoadWeak(weakRef: WeakReferenceProtocol) {
         assertSame(obj, weakDeallocLoadWeak!!.referent)
     }
 }
+
+@Test
+fun testKT41811WithGlobal() {
+    // Attempt to make the state predictable:
+    kotlin.native.internal.GC.collect()
+
+    deallocRetainReleaseDeallocated = false
+    assertFalse(deallocRetainReleaseDeallocated)
+
+    autoreleasepool {
+        {
+            globalDeallocRetainRelease = object: DeallocRetainRelease() {}
+        }()
```

Comment:
The code below should be moved out of the `autoreleasepool`. Otherwise, if the code above actually uses autoreleasepool, the latter can emit a release-ref at the end of the scope, and the object won't be reclaimed by the first GC call.